pkg/discovery/type/git/git.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"strings"
+
+	git "github.com/go-git/go-git/v5"
+	provv1 "github.com/rancher-sandbox/rancheros-operator/pkg/apis/rancheros.cattle.io/v1"
+)
+
+type githubOptions struct {
+	repository string
+	subdir     string
+	ctx        context.Context
+}
+
+type githubSetting func(g *githubOptions) error
+
+// WithRepository sets a git repository to scan releases against
+func WithRepository(s string) githubSetting { //nolint:golint,revive
+	return func(g *githubOptions) error {
+		g.repository = s
+		return nil
+	}
+}
+
+// WithSubpath sets a repository
```

Comment:
incomplete comment? This says nothing about it being the subdir to look for the json files